components/script/dom/gamepad.rs

```diff
@@ -143,10 +193,76 @@ impl Gamepad {
     }
+
+    /// Initialize the number of axes in the "standard" gamepad mapping.
+    /// <https://www.w3.org/TR/gamepad/#dfn-initializing-axes>
+    pub fn init_axes(&self) {
+        let initial_axes: Vec<f64> = vec![
+            0., // Horizontal axis for left stick (negative left/positive right)
+            0., // Vertical axis for left stick (negative up/positive down)
+            0., // Horizontal axis for right stick (negative left/positive right)
+            0., // Vertical axis for right stick (negative up/positive down)
+        ];
+        self.axes
+            .set_data(GlobalScope::get_cx(), &initial_axes)
+            .expect("Failed to set axes data on gamepad.")
+    }
+
+    /// Initialize the number of buttons in the "standard" gamepad mapping.
+    /// <https://www.w3.org/TR/gamepad/#dfn-initializing-buttons>
+    pub fn init_buttons(global: &GlobalScope) -> DomRoot<GamepadButtonList> {
+        let standard_buttons = &[
+            GamepadButton::new(global, false, false), // Bottom button in right cluster
+            GamepadButton::new(global, false, false), // Right button in right cluster
+            GamepadButton::new(global, false, false), // Left button in right cluster
+            GamepadButton::new(global, false, false), // Top button in right cluster
+            GamepadButton::new(global, false, false), // Top left front button
+            GamepadButton::new(global, false, false), // Top right front button
+            GamepadButton::new(global, false, false), // Bottom left front button
+            GamepadButton::new(global, false, false), // Bottom right front button
+            GamepadButton::new(global, false, false), // Left button in center cluster
+            GamepadButton::new(global, false, false), // Right button in center cluster
+            GamepadButton::new(global, false, false), // Left stick pressed button
+            GamepadButton::new(global, false, false), // Right stick pressed button
+            GamepadButton::new(global, false, false), // Top button in left cluster
+            GamepadButton::new(global, false, false), // Bottom button in left cluster
+            GamepadButton::new(global, false, false), // Left button in left cluster
+            GamepadButton::new(global, false, false), // Right button in left cluster
+            GamepadButton::new(global, false, false), // Center button in center cluster
+        ];
+        rooted_vec!(let buttons <- standard_buttons.iter().map(|button| DomRoot::from_ref(&**button)));
+        GamepadButtonList::new(global, buttons.r())
+    }
+
+    #[allow(unsafe_code)]
+    /// <https://www.w3.org/TR/gamepad/#dfn-map-and-normalize-axes>
+    pub fn map_and_normalize_axes(&self, axis_index: usize, value: f64) {
+        let normalized_value: f64 =
+            2.0 * (value - self.axis_bounds.0) / (self.axis_bounds.1 - self.axis_bounds.0) - 1.0;
+        let mut axis_vec = self.axes.get_internal().unwrap();
```

Comment:
let's use `internal_to_option` and warn that axes haven't been initiazed if none.